### ISSUE TITLE
Macro scope issues

### ISSUE THREAD

**Issue by JettMonstersGoBoom** [datatrash/mos#165]
```
  .macro Debug(value) {
    LDA #value
    STA $d020
    STA $d021
  }
  basic_start(start)
start:
  Debug(1)

somescope: {
    Debug(5)
  }
  jmp * 

```
will result in 

"error: unknown identifier: Debug"

inside somescope

**Comment by sagacity** [comment]
Thanks for reporting and the clear reproduction scenario. The upcoming 0.6.0 release has a bunch of improvements related to symbol lookup and thankfully this issue was already fixed on master. I will try to get that out of the door this week and update this issue once that has been done.

**Comment by sagacity** [comment]
Fixed by #162.
